### PR TITLE
Created a cpp program which encrypts/decrypts a text using vignere cipher

### PR DESCRIPTION
**Pull request by him1anshu**
Resolves Issue #193 

### Description:
Created a CPP program that encrypts/decrypts a text using vignere cipher.

The input can have alphanumeric characters and white spaces. It can also have punctuation and special characters. 
The Cipher will only operate on the alphabets though, i.e. [ a-z] and [A-Z].

**Sample Input:**
Enter the key used for encryption : 
fortification
Enter the text for encryption : 
defend the east wall of the castle

**Output:**
Encrypted Text : iswxvibjexiggbocewkbjeviggqs
Decrypted Text : defendtheeastwallofthecastle

Please review the code @Z0Marlin

### CODE REVIEW

**Comment by Z0Marlin** [comment]
The program should retain the whitespaces and other non-alphabetic characters. When encrypting with the key `fortification`, the application should encrypt `defend the east wall of the castle!` and return ` iswxvi bje xigg boce wk bje viggqs!` as the ciphertext.

File: systems/193/vigenere.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#include<iostream>
+#include<string.h>
+using namespace std;
+int main()
+{
+
+int i,keylen,j=0;
```
Comment (Z0Marlin) [inline systems/193/vigenere.cpp]: Please, indent the code.

Comment (him1anshu) [reply systems/193/vigenere.cpp]: ok

**Comment by him1anshu** [comment]
I have done the changes and indentation.@Z0Marlin

File: systems/193/vigenere.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    if (islower(text[i])) {
+      ch = text[i] % 97;
+      dec = ((26 + (ch - (tolower(key[j % keylen])) % 97)) % 26) + 97;
+      text[i] = dec;
+      j++;
+    }
+
+  }
+
+  cout << "Decrypted Text : " << text << "\n";
```
Comment (Z0Marlin) [inline systems/193/vigenere.cpp]: The decrypted text isn't the same as the input.  You must decipher the input using the key and print that.

Comment (him1anshu) [reply systems/193/vigenere.cpp]: Screenshot from 2019-10-05 09-00-07
I am getting the input after deciphering the encrypted text and the input is same as deciphered text.

Comment (him1anshu) [reply systems/193/vigenere.cpp]: Ok

**Comment by him1anshu** [comment]
Made changes to the program so that it takes input of encrypted text by user to decipher it.
Please review .@Z0Marlin

**Comment by him1anshu** [comment]
@SaurabhAgarwala, @niranjanrao1509 please merge.

**Pull request merged**